helion/_compiler/tile_strategy.py

```diff
@@ -433,7 +441,9 @@ def _setup_mask(
             f"{mask_var} = ({index_var} < ({state.device_function.sympy_expr(numel)}))"
         )
 
-    def select_pid_strategy(self) -> ProgramIDs:
+    def select_pid_strategy(self, state: CodegenState) -> ProgramIDs:
+        if (shared_pid := state.device_function.shared_pid) is not None:
+            return shared_pid
         if self.l2_grouping > 1:
             return L2GroupingProgramIDs(group_size=self.l2_grouping)
```

Comment:
Does this work with the l2 grouping stuff?  (It should be possible since L2GroupingProgramIDs only uses a 1D pid.)